contracts/earn/strategies/BankerJoe/StrategyBankerJoeNative.sol

```diff
@@ -0,0 +1,182 @@
+
+    /* ----- Events ----- */
+
+    event StratHarvest(address indexed harvester, uint256 wantHarvested, uint256 tvl);
+    event Deposit(uint256 tvl);
+    event Withdraw(uint256 tvl);
+
+    constructor(
+        address initVault,
+        address initAccessManager,
+        address initIToken,
+        address[] memory initRewardToWantRoute
+    ) public BaseStrategy(initVault, initAccessManager) {
+        iToken = initIToken;
+        rewardToWantRoute = initRewardToWantRoute;
+
+        _giveAllowances();
+    }
+
+    /* ----- External Functions ----- */
+
+    function beforeDeposit() external override {
+        require(msg.sender == address(vault), 'StrategyBankerJoeNative: !vault');
+        if (harvestOnDeposit) {
+            harvest();
+        }
+        updateSupplyBal();
+    }
+
+    function rewardToWant() external view returns(address[] memory) {
+        return rewardToWantRoute;
+    }
+
+    /* ----- Public Functions ----- */
+
+    function harvest() public override whenNotPaused {
+        require(msg.sender == tx.origin || msg.sender == address(vault), 'StrategyBankerJoeNative: EOA_or_vault');
+
+        if (IComptroller(comptroller).pendingImplementation() == address(0)) {
+            uint256 beforeBal = balanceOfWant();
+
+            IComptroller(comptroller).claimReward(0, address(this));
+            IComptroller(comptroller).claimReward(1, address(this));
+            uint256 toWrap = address(this).balance;
+            if (toWrap > 0) {
+                IWETH(wrappedEther).deposit{value: toWrap}();
+            }
+
+            uint256 rewardBal = IERC20(reward).balanceOf(address(this));
+            if (rewardBal > 0) {
+                IJoeRouter(uniRouter).swapExactTokensForTokens(rewardBal, 0, rewardToWantRoute, address(this), now);
+                chargePerformanceFee(balanceOfWant().sub(beforeBal));
+                uint256 wantHarvested = balanceOfWant().sub(beforeBal);
+                deposit();
+
+                lastHarvest = block.timestamp;
+                emit StratHarvest(msg.sender, wantHarvested, balanceOf());
+            }
+        } else {
+            _withdrawAll();
+            pause();
+        }
+    }
+
+    function deposit() public override whenNotPaused nonReentrant {
+        uint256 wantBal = balanceOfWant();
+
+        if (wantBal > 0) {
+            IVToken(iToken).mint(wantBal);
+            updateSupplyBal();
+            emit Deposit(balanceOf());
+        }
+    }
+
+    function withdraw(uint256 amount) public override nonReentrant {
+        require(msg.sender == vault, 'StrategyBankerJoeNative: !vault');
+        require(amount > 0, 'StrategyBankerJoeNative: !amount');
+
+        uint256 wantBal = balanceOfWant();
+
+        if (wantBal < amount) {
+            IVToken(iToken).redeemUnderlying(amount);
+            updateSupplyBal();
+            wantBal = IERC20(want).balanceOf(address(this));
```

Comment:
uint256 newWantBal = IERC20(want).balanceOf(address(this));
   require(newWantBal > wantBal, '....');
   wantBal = newWantBal;